### PR TITLE
Audit hashCode overrides outside of packages/flutter

### PR DESCRIPTION
**Pull request by dnfield**
Replace custom hashCode implementations with Object.hash/hashAll/hashAllUnordered

### CODE REVIEW

File: dev/devicelab/test/adb_test.dart [hunk 1]
```diff
@@ -166,15 +166,14 @@ class CommandArgs {
...
-    : null.hashCode;
-
-  int get _hashEnvironment => environment != null
-    ? const MapEquality<String, String>().hash(environment)
-    : null.hashCode;
+  int get hashCode {
+    return Object.hash(
+      command,
+      Object.hashAll(arguments ?? const <String>[]),
+      Object.hashAllUnordered(environment?.keys ?? const <String>[]),
```
Comment (jonahwilliams) [inline dev/devicelab/test/adb_test.dart:173]: Not that it matters too much, but what does `hashAllUnordered` do here over hash all?

File: dev/tools/vitool/lib/vitool.dart [hunk 2]
```diff
@@ -274,7 +274,7 @@ class FrameData {
   }
 
   @override
-  int get hashCode => size.hashCode ^ paths.hashCode;
+  int get hashCode => Object.hash(size.x, size.y, Object.hashAll(paths));
```
Comment (jonahwilliams) [inline dev/tools/vitool/lib/vitool.dart:277]: Point has a real hashcode https://github.com/dart-lang/sdk/blob/b6f9dbc570c4e9065e922228c9747b4ea8104904/sdk/lib/math/point.dart#L27

File: packages/flutter_tools/lib/src/vscode/vscode.dart [hunk 3]
```diff
@@ -326,5 +326,5 @@ class VsCodeInstallLocation {
 
   @override
   // Lowest bit is for isInsiders boolean.
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/vscode/vscode.dart:328]: is this comment still true?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/vscode/vscode.dart:328]: I guess that is `edition`, kind of confusing

**Review by jonahwilliams — APPROVED** [review]
LGTM with some nits

Thanks for doing this, so much nice now!

File: packages/flutter_tools/lib/src/vscode/vscode.dart (hunk 3 above)
Comment (dnfield) [reply packages/flutter_tools/lib/src/vscode/vscode.dart:328]: This comment didn't make sense to me, so I just left it.

File: dev/tools/vitool/lib/vitool.dart (hunk 2 above)
Comment (dnfield) [reply dev/tools/vitool/lib/vitool.dart:277]: Done

File: dev/devicelab/test/adb_test.dart (hunk 1 above)
Comment (dnfield) [reply dev/devicelab/test/adb_test.dart:173]: Order independent hashing. https://github.com/dart-lang/sdk/commit/83376bf1ee81607a182f8558b242f1c9f4883246#diff-dc1389d26b294bb9804491e03fdf84c939c81eddc77578a0d90417b052f2a3c8R524

It's probably not super necessary here, where the map is always going to be a LinkedHashMap and preserve order, but I did it for the other map so I figured why not here too.

**Pull request merged**